src/wasm-js.cpp

```diff
@@ -69,7 +69,7 @@ extern "C" void EMSCRIPTEN_KEEPALIVE load_asm2wasm(char *input) {
   module = new AllocatingModule();
   module->memory.initial = EM_ASM_INT_V({
     return Module['providedTotalMemory']; // we receive the size of memory from emscripten
-  });
+  }) / Memory::kPageSize;
```

Comment:
Actually it should be validated rather than rounded. It can't be rounded down because then there might not be enough space in linear memory, and it can't be rounded up because the interpreter would think it had more memory than was in the buffer.